File: backend/financial/core/api/src/main/java/br/com/systec/opusfinancial/financial/api/service/AccountService.java
package br.com.systec.opusfinancial.financial.api.service;

import br.com.systec.opusfinancial.financial.api.filter.FilterAccount;
import br.com.systec.opusfinancial.financial.api.domain.Account;
import br.com.systec.opusfinancial.financial.api.domain.TransactionType;
import org.springframework.data.domain.Page;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public interface AccountService {

    Account create(Account account);

    Account update(Account account);

    Page<Account> findByFilter(FilterAccount filter);

    Account findById(UUID id);

    void createDefaultAccount(UUID tenantId);

    List<Account> findByIds(List<UUID> listOfAccountId);

    void updateBalance(UUID accountId, BigDecimal amount, TransactionType transactionType);
}

File: frontend/src/financial/account/service/account.service.ts
import { PageResponse } from '@/common/model/page.response';
import api from '@/service/config/api';
import { AccountInfoResponse } from '../model/account.info.response.model';
import { Account } from '../model/account.model';
import { AccountResponse } from '../model/account.response';

class AccountService {
    private readonly endpoint = '/api/v1/accounts';

    public async findByFilter(): Promise<PageResponse<AccountResponse>> {
        const response = await api.get<PageResponse<AccountResponse>>(`${this.endpoint}/filter`);
        return response.data;
    }

    public async findById(id: string): Promise<Account> {
        const response = await api.get<AccountInfoResponse>(`${this.endpoint}/${id}`);
        return response.data;
    }

    public async save(account: Account): Promise<AccountResponse> {
        if (account.id) {
            return this.update(account);
        }
        return this.create(account);
    }

    private async create(account: Account): Promise<AccountResponse> {
        const response = await api.post<AccountResponse>(this.endpoint, account);
        return response.data;
    }

    private async update(account: Account): Promise<AccountResponse> {
        const response = await api.put<AccountResponse>(`${this.endpoint}/${account.id}`, account);
        return response.data;
    }
}

export default new AccountService();
